### ISSUE TITLE
make equation numbering a default option in mathjax

### ISSUE THREAD

**Issue by jdhao** [theme-next/hexo-theme-next#264]
### I agree and want to create new issue 

- [x ] Yes, I was on [Hexo Docs page](https://hexo.io/docs/), especially on [Templates](https://hexo.io/docs/templates.html), [Variables](https://hexo.io/docs/variables.html), [Helpers](https://hexo.io/docs/helpers.html) and [Troubleshooting](https://hexo.io/docs/troubleshooting.html).
- [ x] Yes, I was on [NexT Documentation Site](http://theme-next.org/docs/).
- [ x] And yes, I already searched for current [issues](https://github.com/theme-next/hexo-theme-next/issues?utf8=%E2%9C%93&q=is%3Aissue) and this is not help to me.

### suggestions

Right now I am using `hexo-renderer-pandoc` to render latex equations in my post. But in order to support equation numbering, as is shown [in this post](https://jdhao.github.io/2018/01/25/hexo-mathjax-equation-number/), we need to change the `mathjax.swig` file under the theme next root directory (the path is `layout\_third-party\math\mathjax.swig`).  We need to change 

```
<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"]  ],
        processEscapes: true,
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
      }
    });
</script>
```
to 

```
<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"]  ],
        processEscapes: true,
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
      },
      TeX: {equationNumbers: { autoNumber: "AMS" }}
    });
</script>
```

in order to support automatic equation numbering. I hope the following option may be added as a default option:

```
 TeX: {equationNumbers: { autoNumber: "AMS" }}
```

**Comment by jdhao** [comment]
yep, I have read it. But it has nothing to do what I propose here.

**Comment by ivan-nginx** [comment]
Ok, can u give some examples here with *.md codes and screenshoots maybe? Where is the difference?

**Comment by jdhao** [comment]
Without the equation numbering option. We can not refer to equations easily in our post. For example, for the following equation, 

```
$$\begin{equation}
e=mc^2
\end{equation}\label{eq1}$$
```

we can later easily refer to it. A sample markdown snippet may be:

```
the famous matter-energy equation $\eqref{eq1}$ is proposed by Einstein.
```

With that option, the equation will be numbered and correctly referred in the text in a concise way.

**Comment by ivan-nginx** [comment]
Oh, it's like variables, right? Can u give full `*.md` content to reproduce this?

**Comment by jdhao** [comment]
I have posted a full markdown file [here](https://gist.githubusercontent.com/jdhao/9806559082a53547549af4dc18a7ccd6/raw/bf061baf156fc013c7aad676d5f2c389a9898c68/markdown_latex_equation_numbering.md). It is used to generate [post here](https://jdhao.github.io/2018/01/25/hexo-mathjax-equation-number/).

**Comment by ivan-nginx** [comment]
Ok, so, i get it after reproduce content in [comment above](https://github.com/theme-next/hexo-theme-next/issues/264#issuecomment-384848368):

1. As is for now from master branch:
    ![image](https://user-images.githubusercontent.com/16944225/39343083-bc8494ea-49e3-11e8-9a0f-7863b73c1c66.png)
2. With `TeX: {equationNumbers: { autoNumber: "AMS" }}` additions: 
    ![image](https://user-images.githubusercontent.com/16944225/39343053-89ed22d6-49e3-11e8-953c-307e8aaf1c8e.png)

So, i think it can be added by default without any breaking changes in MathJax code.
@jdhao u can to do it, i think. Also, it would be great if u update [Math docs in EN/CN](https://github.com/theme-next/hexo-theme-next/issues/264#issuecomment-384843919) language.

**Comment by sli1989** [comment]
@jdhao Do you have time to do this PR?

**Comment by sli1989** [comment]
1. take your time , you can take [this](https://help.github.com/articles/creating-a-pull-request/) for reference.
1. remember to update the [Math docs in EN/CN](https://github.com/theme-next/hexo-theme-next/issues/264#issuecomment-384843919). add some instruction of how to use it.

**Comment by jdhao** [comment]
OK, I will try to make a pull request once it is done.

**Comment by jdhao** [comment]
@sli1989 ，I have made a pull request. Can you please review it?

**Comment by sli1989** [comment]
Hi, [gitter](https://gitter.im/theme-next) and [telegram](https://t.me/theme_next) chat are here, welcome to join in.

**Comment by jdhao** [comment]
Joined gitter chat :)
